code/modules/reagents/chemistry/reagents/drug_reagents.dm

```diff
@@ -361,3 +361,84 @@
 		M.adjustOxyLoss(1, 0)
 	..()
 	. = 1
+
+/datum/reagent/drug/happiness
+	name = "Happiness"
+	id = "happiness"
+	description = "Fills you with ecstasic numbness and causes minor brain damage. Highly addictive. If overdosed causes sudden mood swings."
+	reagent_state = LIQUID
+	color = "#FFF378"
+	addiction_threshold = 10
+	overdose_threshold = 20
+
+/datum/reagent/drug/happiness/on_mob_add(mob/living/L)
+	..()
+	L.add_trait(TRAIT_FEARLESS, id)
+	SEND_SIGNAL(L, COMSIG_ADD_MOOD_EVENT, "happiness_drug", /datum/mood_event/happiness_drug)
+
+/datum/reagent/drug/happiness/on_mob_delete(mob/living/L)
+	L.remove_trait(TRAIT_FEARLESS, id)
+	SEND_SIGNAL(L, COMSIG_CLEAR_MOOD_EVENT, "happiness_drug")
+	..()
+
+/datum/reagent/drug/happiness/on_mob_life(mob/living/carbon/M)
+	M.jitteriness = 0
+	M.confused = 0
+	M.disgust = 0
+	M.adjustBrainLoss(0.2)
+	..()
+	. = 1
+
+/datum/reagent/drug/happiness/overdose_process(mob/living/M)
+	if(prob(30))
+		var/reaction = rand(1,3)
+		switch(reaction)
+			if(1)
+				M.emote("laugh")
+				SEND_SIGNAL(M, COMSIG_ADD_MOOD_EVENT, "happiness_drug", /datum/mood_event/happiness_drug_good_od)
+			if(2)
+				M.emote("sway")
+				M.Dizzy(25)
+			if(3)
+				M.emote("frown")
+				SEND_SIGNAL(M, COMSIG_ADD_MOOD_EVENT, "happiness_drug", /datum/mood_event/happiness_drug_bad_od)
```

Comment:
Maybe you could have these OD moodlets cancel each other whenever they're applied (i.e. when you get the bad one, the good one immediately goes away (and the bad one goes into effect), and when you get the good one, the bad one immediately goes away (and the good one goes into effect)).

Reply:
that's how it works

Reply:
Oh, my bad, then.

Reply:
Wait, what I meant was not that their modifiers to mood would negate each other, but rather that if you had both the good overdose and bad overdose moodlets at the same time, you'd have a mood (bar other modifiers) of either +100 or -100 (NOT 0), depending on which one was applied most recently.

Reply:
The bad moodlet takes 30 seconds to go away, but as far as I can tell, the good and the bad moodlets each have a 10% chance of going into effect each second. Thus, after the first few seconds, your mood should be mostly stable, as you'll almost always have both the bad and the good overdose moodlets.

Reply:
The moodlets override each other, you can't have both at the same time

Reply:
Oh, okay.